addons/binding/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/handler/EnergenieSubdevicesHandler.java

```diff
@@ -0,0 +1,655 @@
+                this.gatewayHandler = (EnergenieGatewayHandler) handler;
+            } else {
+                throw new IllegalStateException(
+                        "BridgeHandler should be of type " + EnergenieGatewayHandler.class.getName());
+            }
+            // Check the bridge status
+            ThingStatus bridgeStatus = bridge.getStatus();
+            if (bridgeStatus == ThingStatus.ONLINE) {
+                initializeInternal();
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                        "Bridge status is " + bridgeStatus.toString());
+            }
+        } else {
+            logger.warn("Can't initialize ThingHandler, bridge is missing");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR, "Bridge is missing");
+        }
+
+    }
+
+    private void initializeInternal() {
+        // Bridge is ONLINE and this should not be the case, we simply double check
+        if (gatewayHandler == null || !isGatewayReady()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                    "Bridge isn't set up correctly");
+            return;
+        }
+
+        Integer deviceID = getDeviceID();
+        if (deviceID != null) {
+            logger.debug("Device with ID {} is already paired", deviceID);
+
+            this.energenieID = deviceID;
+
+            JsonObject data = gatewayHandler.getSubdeviceData(this.energenieID);
+            if (data != null) {
+                updateThingProperties(data);
+
+                updateStatus(ThingStatus.ONLINE);
+
+                scheduleRegularUpdate(updateInterval);
+            } else {
+                logger.debug("Mi|Home server may be temporary unavailable or device with ID {} may be deleted.",
+                        this.energenieID);
+                updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                        "Missing data for paired device with ID " + this.energenieID);
+            }
+        } else {
+            schedulePairing(gatewayHandler.getGatewayId());
+        }
+    }
+
+    private boolean isGatewayReady() {
+        return gatewayHandler.getGatewayId() > 0;
+    }
+
+    private Integer getDeviceID() {
+        Integer id = null;
+        // The thing has been once initialized and the deviceID has been persisted
+        logger.debug("Searching deviceID in the thing configuration.");
+        Configuration config = getConfig();
+        BigDecimal deviceID = (BigDecimal) config.get(EnergenieBindingConstants.PROPERTY_DEVICE_ID);
+        if (deviceID != null) {
+            id = deviceID.intValue();
+            return id;
+        }
+
+        // The thing has been created by a DiscoveryService and the deviceID is included in the properties
+        logger.debug("Searching device ID in the thing properties.");
```

Comment:
I am not sure if I understand your concern correctly.

I will try to explain the situation.

The deviceID is not added to the configuration, as it is internal API information, the user should not have control on it (and should not care at all). What I am trying to do here is a little bit dirty trick - add it as a configuration property, without adding it in the config description - this will make it invisible to the user in the PaperUI. I would not like to keep it simply as property, as properties are not persisted... But this dirty trick does not work with the DiscoveryService, as it simply looks at the config description before deciding if this is a configuration property or a thing property. Maybe you can help with an advice, what to do in this situation.